api/tests/integration/infrastructure/plugins/pino_test.js

```diff
@@ -0,0 +1,137 @@
+const config = require('../../../../lib/config');
+const { expect, HttpTestServer, generateValidRequestAuthorizationHeader } = require('../../../test-helper');
+const pinoPlugin = require('../../../../lib/infrastructure/plugins/pino');
+const monitoringTools = require('../../../../lib/infrastructure/monitoring-tools');
+
+function sink(func) {
+  const result = split(JSON.parse);
+  result.pipe(writeStream.obj(func));
+  return result;
+}
+
+describe('Integration | Infrastructure | plugins | pino', function () {
+  let httpTestServer;
+
+  beforeEach(async function () {
+    httpTestServer = new HttpTestServer();
+    await httpTestServer.register({
+      register: (server) => {
+        server.route([
+          {
+            method: 'GET',
+            path: '/',
+            config: {
+              handler: () => {
+                monitoringTools.incrementInContext('metrics.knexQueryCount');
+                return { cou: 'cou' };
+              },
+            },
+          },
+        ]);
+      },
+      name: 'test-api',
+    });
+  });
+
+  async function registerWithPlugin(cb) {
+    const pinoPluginWithLogger = {
+      ...pinoPlugin,
+      options: {
+        ...pinoPlugin.options,
+        instance: undefined,
+        level: 'info',
+        stream: sink(cb),
+      },
+    };
+    await httpTestServer.register([pinoPluginWithLogger]);
+  }
+
+  context('with request monitoring disabled', function () {
+    let originalStatus;
+    beforeEach(function () {
+      originalStatus = config.hapi.enableRequestMonitoring;
+      config.hapi.enableRequestMonitoring = false;
+      monitoringTools.installHapiHook();
+    });
+
+    afterEach(function () {
+      config.hapi.enableRequestMonitoring = originalStatus;
+    });
+
+    it('should log version', async function () {
+      // given
+      let finish;
+
+      const done = new Promise(function (resolve) {
+        finish = resolve;
+      });
+      const messages = [];
+      await registerWithPlugin((data) => {
+        messages.push(data);
+        finish();
+      });
+
+      const method = 'GET';
+      const url = '/';
+
+      // when
+      const response = await httpTestServer.request(method, url);
+      await done;
+
+      // then
+      expect(response.statusCode).to.equal(200);
+      expect(messages).to.have.lengthOf(1);
+      expect(messages[0].req.version).to.equal('development');
+      expect(messages[0].req.user_id).to.be.undefined;
+      expect(messages[0].req.route).to.be.undefined;
+      expect(messages[0].req.metrics).to.be.undefined;
+    });
+  });
+
+  context('with request monitoring enabled', function () {
+    let originalStatus;
+    beforeEach(function () {
+      originalStatus = config.hapi.enableRequestMonitoring;
+      config.hapi.enableRequestMonitoring = true;
+      monitoringTools.installHapiHook();
+    });
+
+    afterEach(function () {
+      config.hapi.enableRequestMonitoring = originalStatus;
+    });
```

Comment:
Question: pourquoi creer un beforeEach pour un seul test?
Suggestion: utiliser sinon pour ne pas avoir à gerer l'after each.
```suggestion
    beforeEach(function () {
      sinon.stub(config.hapi, 'enableRequestMonitoring').value(true);
      monitoringTools.installHapiHook();
    });
```

Reply:
J'avais mis dans un beforeEach pour m'assurer que l'état serai bien restauré ensuite. 
Trop bien pour sinon, j'ai mis à jour ça.